### PR TITLE
Allow ipaapi user to access SSSD's info pipe

### PR DESCRIPTION
**Pull request by tiran**
For smart card authentication, ipaapi must be able to access to sss-ifp.
During installation and upgrade, the ipaapi user is now added to
[ifp]allowed_uids.

The commit also fixes two related issues:

* The server upgrade code now enables ifp service in sssd.conf. The
  existing code modified sssd.conf but never wrote the changes to disk.
* sssd_enable_service() no longer fails after it has detected an
  unrecognized service.

Fixes: https://pagure.io/freeipa/issue/7751
Signed-off-by: Christian Heimes <cheimes@redhat.com>

### CODE REVIEW

File: ipaserver/install/server/upgrade.py [hunk 1]
```diff
@@ -2017,15 +2033,8 @@ def upgrade_configuration():
...
-    set_sssd_domain_option('ipa_server_mode', 'True')
-    set_sssd_domain_option('ipa_server', api.env.host)
 
-    sssdconfig = SSSDConfig.SSSDConfig()
-    sssdconfig.import_config()
-    sssd_enable_service(sssdconfig, 'ifp')
-
-    sssd = services.service('sssd', api)
-    sssd.restart()
+    sssd_updgrade()
```
Comment (rcritten) [inline ipaserver/install/server/upgrade.py]: Typo?

File: ipaserver/install/server/upgrade.py [hunk 2]
```diff
@@ -1408,6 +1408,22 @@ def set_sssd_domain_option(option, value):
     sssdconfig.write(paths.SSSD_CONF)
 
 
+def sssd_updgrade():
```
Comment (rcritten) [inline ipaserver/install/server/upgrade.py]: updgrade?

File: ipaserver/install/server/upgrade.py (hunk 1 above)
Comment (tiran) [reply ipaserver/install/server/upgrade.py]: Yeah, I couldn't decide on either update or upgrade, so I wrote updgrade.

**Comment by tiran** [comment]
master:

* 8b0f3595fd94a31a0e2d5a83dc74a430c973325b Allow ipaapi user to access SSSD's info pipe

**Pull request closed without merging**